__mocks__/react-native-youtube-iframe.ts

```diff
@@ -10,19 +12,32 @@ const YouTubePlayerMock = React.forwardRef(function Component(
     onChangeState?: (state: string) => void
     onError?: () => void
   },
-  ref: ForwardedRef<Partial<YoutubeIframeRef>>
+  ref: ForwardedRef<YoutubeIframeRef>
 ) {
   React.useEffect(() => {
     if (props.onReady) props.onReady()
     if (props.onChangeState) props.onChangeState(mockState)
     if (props.onError && mockError) props.onError()
   }, [props])
 
+  if (typeof ref === 'object' && ref !== null) {
+    ref.current = {
+      getCurrentTime: jest.fn().mockReturnValue(134.9),
+      getDuration: jest.fn().mockReturnValue(267.4),
+      getVideoUrl: jest.fn(),
+      isMuted: jest.fn(),
+      getVolume: jest.fn(),
+      getPlaybackRate: jest.fn(),
+      getAvailablePlaybackRates: jest.fn(),
+      seekTo: jest.fn(),
+    }
+  }
+
   // @ts-ignore avoid internal typing complexity
-  return React.createElement(YouTubePlayer, { ref, ...props })
+  return React.createElement(YouTubePlayer, props)
 })
 
-const setPlayerState = (playerState: PLAYER_STATES) => {
+const setPlayerState = (playerState: PlayerState) => {
```

Comment:
le playerState ici simule l'état interne de la lib, je pense que c'est mieux de reprendre l'enum venant de la lib

Reply:
les tests passent plus avec ce typage :  TypeError: Cannot read properties of undefined (reading 'ENDED')

Reply:
bizarre, on pourra voir ça ensemble si tu veux